Flaky test: test_coordinator_save_state_format fails due to EMA floating-point precision

## Summary

`test_coordinator_save_state_format` in `tests/scenarios/test_coordinator_persistence.py` is flaky — it fails intermittently (~25% of runs) with a floating-point precision error.

## Failure

```
>       assert saved_data["zones"]["zone1"]["current"] == 21.5
E       assert 21.500000000000004 == 21.5

tests/scenarios/test_coordinator_persistence.py:172: AssertionError
```

## Root Cause

The test manually sets `runtime.state.current = 21.5`, then calls `coordinator.async_refresh()` (line 141). During refresh, `_update_zone()` reads the temperature sensor entity (also `"21.5"`) and applies EMA smoothing via `apply_ema()`:

```python
# core/ema.py:36-39
alpha = dt / (tau + dt)
return alpha * current + (1 - alpha) * previous
```

When both `current` and `previous` are `21.5`, the result should be exactly `21.5`. However, certain `alpha` values cause IEEE 754 floating-point rounding to produce `21.500000000000004` instead.

The flakiness comes from `dt` varying between runs. The coordinator calculates `dt = (now - last_update_success_time).total_seconds()`, which depends on how much wall-clock time elapses between the two `async_refresh()` calls (lines 117 and 141). Different `dt` values produce different `alpha` values, and some trigger the floating-point error while others don't.

### Reproduction

```bash
for i in $(seq 1 20); do
  uv run pytest tests/scenarios/test_coordinator_persistence.py::test_coordinator_save_state_format -x -q 2>&1 | tail -1
done
```

Typically fails ~5 out of 20 runs.

## Suggested Fix

Use `pytest.approx()` for the floating-point comparison:

```python
# Line 172
assert saved_data["zones"]["zone1"]["current"] == pytest.approx(21.5)
```

Alternatively, freeze time (e.g., with `freezegun` / `pytest_freezer`, which is already a test dependency) so that `dt` is deterministic:

```python
from freezegun import freeze_time

@freeze_time("2026-01-15 12:00:00", tz_offset=0)
async def test_coordinator_save_state_format(...):
    ...
```

The `pytest.approx` fix is simpler and directly addresses the symptom. Freezing time is a more robust fix that eliminates all timing-dependent behavior in the test.

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -247,6 +247,12 @@ def test_duty_cycle_edge_case():
     assert calculate_duty_cycle(edge_case) == expected
 ```
 
+### 5. Floating-Point Equality in Tests
+Use `pytest.approx()` for asserting float values that have passed through
+any computation (EMA, PID, division, interpolation). Direct assign-and-read
+without computation can use `==`. Dictionary comparisons with nested floats
+are especially dangerous — assert float fields individually with `approx()`.
+
 ## CI Workflows
 
 The CI runs three workflow files on PRs:

--- a/tests/scenarios/test_coordinator_persistence.py
+++ b/tests/scenarios/test_coordinator_persistence.py
@@ -3,6 +3,7 @@
 from datetime import UTC, datetime, timedelta
 from unittest.mock import AsyncMock, patch
 
+import pytest
 from homeassistant.core import HomeAssistant
 from pytest_homeassistant_custom_component.common import MockConfigEntry
 
@@ -162,20 +163,20 @@ async def test_coordinator_save_state_format(
     assert "last_update_success_time" in saved_data
 
     # Verify full PID state is saved
-    assert saved_data["zones"]["zone1"]["pid_error"] == 1.5
-    assert saved_data["zones"]["zone1"]["pid_proportional"] == 25.0
-    assert saved_data["zones"]["zone1"]["pid_integral"] == 75.0
-    assert saved_data["zones"]["zone1"]["pid_derivative"] == 0.8
-    assert saved_data["zones"]["zone1"]["duty_cycle"] == 65.0
+    assert saved_data["zones"]["zone1"]["pid_error"] == pytest.approx(1.5)
+    assert saved_data["zones"]["zone1"]["pid_proportional"] == pytest.approx(25.0)
+    assert saved_data["zones"]["zone1"]["pid_integral"] == pytest.approx(75.0)
+    assert saved_data["zones"]["zone1"]["pid_derivative"] == pytest.approx(0.8)
+    assert saved_data["zones"]["zone1"]["duty_cycle"] == pytest.approx(65.0)
 
     # Verify EMA temperature is saved (V2 uses 'current' key)
-    assert saved_data["zones"]["zone1"]["current"] == 21.5
+    assert saved_data["zones"]["zone1"]["current"] == pytest.approx(21.5)
     # Verify display temperature is saved
-    assert saved_data["zones"]["zone1"]["display_temp"] == 21.5
+    assert saved_data["zones"]["zone1"]["display_temp"] == pytest.approx(21.5)
     # Verify preset_mode is saved
     assert saved_data["zones"]["zone1"]["preset_mode"] == "eco"
     # Verify used_duration is saved
-    assert saved_data["zones"]["zone1"]["used_duration"] == 1234.5
+    assert saved_data["zones"]["zone1"]["used_duration"] == pytest.approx(1234.5)
 
 
 async def test_coordinator_handles_invalid_timestamp_format(

